docs/element-definitions.md

```diff
@@ -234,6 +239,49 @@ Note: At the moment only the HTTP protocol is supported.
 }
 ```
 
+### Affordance (Base API Element)
+
+An affordance element provides a way to provide affordance definitions for the
+API. These definitions can describe link relations used in a Transition Element,
+along with defining unsafe and idempotent transitions.
+
+While data structures are used for defining semantics around data models,
+affordances are used for defining semantics around hypermedia state transitions.
+
+The control types used here are based on the types used in [ALPS][]. The meta
+property `id` SHOULD be used to define the machine-usable name of the
+affordance.
+
+#### Properties
+
+- `element`: affordance
+- `attributes`
+    - `controlType` (enum)
+        - safe - A hypermedia control that triggers a safe, idempotent state
+          transition (e.g. HTTP.GET or HTTP.HEAD).
+        - unsafe - A hypermedia control that triggers an unsafe, non-idempotent
+          state transition (e.g. HTTP.POST).
+        - idempotent - A hypermedia control that triggers an unsafe, idempotent
```

Comment:
@pksunkara in my email, I listed two options we have in addition to the ALPS way:

1. An array of enum (like you said)
1. Two boolean attributes: one for safety, one for idempotency

Another option is to make up our own terms and categories. I just mention that here because it might be something good to just think about. However I can't come up with anything right now :)